numpy/linalg/_linalg.py

```diff
@@ -721,31 +721,39 @@ def matrix_power(a, n):
 
 # Cholesky decomposition
 
+def _cholesky_dispatcher(a, /, *, upper=None):
+    return (a,)
 
-@array_function_dispatch(_unary_dispatcher)
-def cholesky(a):
+
+@array_function_dispatch(_cholesky_dispatcher)
+def cholesky(a, /, *, upper=False):
     """
     Cholesky decomposition.
 
-    Return the Cholesky decomposition, `L * L.H`, of the square matrix `a`,
-    where `L` is lower-triangular and .H is the conjugate transpose operator
-    (which is the ordinary transpose if `a` is real-valued).  `a` must be
-    Hermitian (symmetric if real-valued) and positive-definite. No
-    checking is performed to verify whether `a` is Hermitian or not.
-    In addition, only the lower-triangular and diagonal elements of `a`
-    are used. Only `L` is actually returned.
+    Return the lower or upper Cholesky decomposition, `L * L.H` or `U.H * U`,
+    of the square matrix `a`, where `L` is lower-triangular, `U` is
+    upper-triangular, and .H is the conjugate transpose operator (which is
+    the ordinary transpose if `a` is real-valued). `a` must be Hermitian
+    (symmetric if real-valued) and positive-definite. No checking is performed
+    to verify whether `a` is Hermitian or not. In addition, only the lower or
+    upper-triangular and diagonal elements of `a` are used. Only `L` or `U`
+    is actually returned.
```

Comment:
I think all of these should be double backtick quotes, since they're not cross-references to numpy types.

Reply:
Right, I formatted them.